Automattic/woocommerce-payments
# Normalize HK addresses for Apple Pay

Fixes https://github.com/Automattic/woocommerce-payments/issues/9123

#### Changes proposed in this Pull Request.

This PR is a port of the WC Stripe implementation for the same issue. Please refer to the [PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1593) on their repo for more information.

Basically, the changes introduced are as follows:

>Instead of using the billing_state or shipping_state to normalize the region for a Hong Kong address, we instead use the billing_postcode and shipping_postcode which seem to contain the equivalent for Hong Kong addresses.










#### Testing instructions

1. Add the following address to your Apple Pay shipping options:
    - Country: `Hong Kong`
    - Region: `New Territories`
    - District: `Taipo`
    - Address: `9 Feng Yuen Road`
2. `git checkout develop`.
3. Add any product to your cart.
4. Attempt to checkout via Apple Pay using the address you just entered as a billing address and/or shipping address.
5. Notice that you are unable to checkout and no WooCommerce order is created.
6. `git checkout as-hk-address`
7. Try checking out via Apple Pay using the Hong Kong address.
8. Notice that you can now check out successfully.

-------------------


**Post merge**
